lib/aws/lex/conversation/type/base.rb

```diff
@@ -27,7 +27,7 @@ def assign_attributes!(opts = {})
 
             def to_lex
               self.class.attributes.each_with_object({}) do |attribute, hash|
-                value = transform_to_lex(instance_variable_get("@#{attribute}"))
+                value = transform_to_lex(public_send(attribute))
```

Comment:
What's the reasoning behind this change? Perhaps we have attribute accessor methods that don't map back to an underlying instance variable?

Reply:
you bet! slots is the culprit. It only gets populated when it's called as a method.

Reply:
Got it, sounds good to me then!